src/Components/Body/Chat/LeftFold.tsx

```diff
@@ -0,0 +1,70 @@
+import { MdOutlineExpandMore } from "react-icons/md";
+import { IoFilterOutline } from "react-icons/io5";
+import { RiChatNewLine } from "react-icons/ri";
+import { RxTriangleDown, RxTriangleRight } from "react-icons/rx";
+import ChatBar from "./ChatBar";
+import { ChatListDataType } from "../../../Types/types";
+import { useState } from "react";
+type ChatLeftFoldProps = {
+  chatList: ChatListDataType[];
+  selectedChat: number;
+  changeSelectedChat: (num: number) => void
+}
+export default function ChatLeftFold({
+  chatList,
+  selectedChat,
+  changeSelectedChat
+}: ChatLeftFoldProps) {
+  const [showPinned,setShowPinned]= useState<boolean>(true)
+  const [showRecent,setShowRecent]= useState<boolean>(true)
```

Comment:
Can create a custom hook `useToggleVisibility` which will return current state and a toggle function.